python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -484,32 +484,48 @@ def do_evaluate(
             return self._distinct(
                 column,
                 keep=plc.stream_compaction.DuplicateKeepOption.KEEP_FIRST,
-                source_value=plc.interop.from_arrow(pa.scalar(True)),  # noqa: FBT003
-                target_value=plc.interop.from_arrow(pa.scalar(False)),  # noqa: FBT003
+                source_value=plc.interop.from_arrow(
+                    pa.scalar(True, type=plc.interop.to_arrow(self.dtype))  # noqa: FBT003
```

Comment:
can we pacify ruff by doing 
```suggestion
                    pa.scalar(value=True, type=plc.interop.to_arrow(self.dtype))
```
here and elsewhere?

Reply:
Good idea, done.